syntax = "proto3";

import "google/protobuf/empty.proto";
package namespace_proto;

message Ok {
  bool ok = 1;
}

message GetOk {
  bool ok = 1;
}

message CallMe{
  bool need_call = 1;
}

message User{
  int32 id = 1;
  int32 rate = 2;
}

message Unit{
  int32 id_unit = 1;
  int32 amount_unit = 2;
  int32 sum_of_health = 3;
}

message Cell{
  int32 x = 1;
  int32 y = 2;
  int32 id_hero = 3;
  Unit unit = 4;
}

message EnableCell{
  repeated Cell cells = 1;
}

message GameBoard{
  repeated Cell game_board = 1;
}

message FromTo{
  Cell start = 1;
  Cell finish = 2;
  GameId id = 3;
}

message GameId{
  int32 game_id = 1;
}

service Server {
  rpc SendOk(GetOk) returns (Ok) {}
  rpc CallClient(google.protobuf.Empty) returns(stream CallMe) {}
  rpc SendGameBoardToSecond(google.protobuf.Empty) returns(GameBoard) {}
  rpc ReceiveGameBoard(GameId) returns(GameBoard) {}
  rpc AddToLobby(User) returns(GameId) {}
  rpc MoveFromTo(FromTo) returns(GameBoard) {}
}
